addons/pkg-npm/jaxrs/src/main/java/org/commonjava/indy/pkg/npm/jaxrs/NPMContentAccessHandler.java

```diff
@@ -0,0 +1,498 @@
+                response = formatResponse( e, builderModifier );
+            }
+        }
+
+        logger.info( "RETURNING RESULT: {}:{}", sk, path );
+        return response;
+    }
+
+    private void generateNPMContentsFromTransfer( final Transfer transfer, final EventMetadata eventMetadata,
+                                                  Response response,
+                                                  final Consumer<Response.ResponseBuilder> builderModifier )
+    {
+        if ( transfer == null || !transfer.exists() )
+        {
+            return;
+        }
+
+        Transfer versionTarget;
+        Transfer tarballTarget;
+        String versionContent = "";
+        String tarballContent = "";
+
+        ConcreteResource resource = transfer.getResource();
+        try
+        {
+            ObjectMapper mapper = new ObjectMapper();
+            JsonNode root = mapper.readTree( transfer.openInputStream( true ) );
+
+            String versionPath = null;
+            String tarballPath = null;
+            JsonNode vnode = root.path( "versions" );
+            JsonNode anode = root.path( "_attachments" );
+            JsonNode idnode = root.path( "_id" );
+
+            if ( vnode.fields().hasNext() )
+            {
+                Map.Entry<String, JsonNode> entry = vnode.fields().next();
+                String version = entry.getKey();
+                if ( version == null )
+                {
+                    return;
+                }
+                versionPath = Paths.get( idnode.asText(), version ).toString();
+                versionContent = entry.getValue().toString();
+            }
+
+            if ( anode.fields().hasNext() )
+            {
+                String tarball = anode.fields().next().getKey();
+                if ( tarball == null )
+                {
+                    return;
+                }
+                tarballPath = Paths.get( idnode.asText(), "-", tarball ).toString();
+                tarballContent = anode.findPath( "data" ).asText();
+            }
+
+            if ( versionPath == null || tarballPath == null )
+            {
+                return;
+            }
+            versionTarget = transfers.getCacheReference( new ConcreteResource( resource.getLocation(), versionPath ) );
+            logger.info( "STORE {}", versionTarget.getResource() );
+
+            tarballTarget = transfers.getCacheReference( new ConcreteResource( resource.getLocation(), tarballPath ) );
+            logger.info( "STORE {}", tarballTarget.getResource() );
+        }
+        catch ( final IOException e )
+        {
+            logger.error( String.format( "[NPM] Json node parse failed for resource: %s. Reason: %s", resource,
+                                         e.getMessage() ), e );
+            response = formatResponse( e, builderModifier );
+
+            return;
+        }
+
+        if ( versionTarget == null || tarballTarget == null )
```

Comment:
these versionTarget and tarballTarget will always be not null as you have called the getResource on both in the previous logger.info